gno.land/pkg/gnoland/app_test.go

```diff
@@ -1,20 +1,173 @@
 package gnoland
 
 import (
+	"context"
 	"errors"
 	"fmt"
+	"path/filepath"
 	"strings"
 	"testing"
+	"time"
 
-	"github.com/gnolang/gno/gnovm/stdlibs/std"
+	"github.com/gnolang/gno/gno.land/pkg/sdk/vm"
+	"github.com/gnolang/gno/gnovm/pkg/gnoenv"
+	gnostd "github.com/gnolang/gno/gnovm/stdlibs/std"
+	"github.com/gnolang/gno/tm2/pkg/amino"
 	abci "github.com/gnolang/gno/tm2/pkg/bft/abci/types"
-	"github.com/gnolang/gno/tm2/pkg/bft/types"
+	bft "github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/crypto"
+	"github.com/gnolang/gno/tm2/pkg/db/memdb"
 	"github.com/gnolang/gno/tm2/pkg/events"
+	"github.com/gnolang/gno/tm2/pkg/log"
 	"github.com/gnolang/gno/tm2/pkg/sdk"
+	"github.com/gnolang/gno/tm2/pkg/std"
+	"github.com/gnolang/gno/tm2/pkg/store"
+	"github.com/gnolang/gno/tm2/pkg/store/dbadapter"
+	"github.com/gnolang/gno/tm2/pkg/store/iavl"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+// Tests that NewAppWithOptions works even when only providing a simple DB.
+func TestNewAppWithOptions(t *testing.T) {
+	app, err := NewAppWithOptions(&AppOptions{
+		DB: memdb.NewMemDB(),
+		InitChainerConfig: InitChainerConfig{
+			CacheStdlibLoad:        true,
+			GenesisTxResultHandler: PanicOnFailingTxResultHandler,
+			StdlibDir:              filepath.Join(gnoenv.RootDir(), "gnovm", "stdlibs"),
+		},
+	})
+	require.NoError(t, err)
+	bapp := app.(*sdk.BaseApp)
+	assert.Equal(t, "dev", bapp.AppVersion())
+	assert.Equal(t, "gnoland", bapp.Name())
+
+	addr := crypto.AddressFromPreimage([]byte("test1"))
+	resp := bapp.InitChain(abci.RequestInitChain{
+		Time:    time.Now(),
+		ChainID: "dev",
+		ConsensusParams: &abci.ConsensusParams{
+			Block: &abci.BlockParams{
+				MaxTxBytes:   1_000_000,   // 1MB,
+				MaxDataBytes: 2_000_000,   // 2MB,
+				MaxGas:       100_000_000, // 100M gas
+				TimeIotaMS:   100,         // 100ms
+			},
+		},
+		Validators: []abci.ValidatorUpdate{},
+		AppState: GnoGenesisState{
+			Balances: []Balance{
+				{
+					Address: addr,
+					Amount:  []std.Coin{{Amount: 1e15, Denom: "ugnot"}},
+				},
+			},
+			Txs: []std.Tx{
+				{
+					Msgs: []std.Msg{vm.NewMsgAddPackage(addr, "gno.land/r/demo", []*std.MemFile{
+						{
+							Name: "demo.gno",
+							Body: "package demo; func Hello() string { return `hello`; }",
+						},
+					})},
+					Fee:        std.Fee{GasWanted: 1e6, GasFee: std.Coin{Amount: 1e6, Denom: "ugnot"}},
+					Signatures: []std.Signature{{}}, // one empty signature
+				},
+			},
+		},
+	})
+
+	if !resp.IsOK() {
+		t.Fatal(resp)
+	}
+
+	tx := amino.MustMarshal(std.Tx{
```

Comment:
Why panic? Use the normal marshal and handle the error in this test

Reply:
Why do you care when we do it here, for a message that is always the same and unless amino's screwed up won't panic, anyway, when we literally use amino.MustMarshal everywhere in the code, even in places where we shouldn't (ie. non-testing code)?